### DISCUSSION TITLE
Getting wrong coordinates after croping page

### DISCUSSION THREAD

**Discussion by Kavan72** [pymupdf/PyMuPDF#1272] [Q&A] [answered]
I tried to crop the page using set_cropbox method and it's worked perfect. But the problem is when i tried to get cordinate from different library(pdfplumber, pypdf2) then i am getting different coordinate.

## Reproduce Steps
```
# cropped pdf usign PyMupdf library 

import fitz

page_number = 0
doc = fitz.open('test.pdf')
page = doc[page_number].set_cropbox(rect=(64, 100, 500, 300))

doc.select([page_number])
doc.save('data.pdf', garbage=3, deflate=True)

# trying to get cordinate using pdfplumber library
import pdfplumber

with pdfplumber.open("data.pdf") as pdf:
    page = pdf.pages[0]
    print(page.cropbox)  # output:- [Decimal('64'), Decimal('492'), Decimal('500'), Decimal('692')] 

# trying to get cordinate using pypdf2 library
from PyPDF2 import PdfFileReader

with open("data.pdf", "rb") as in_f:
    input1 = PdfFileReader(in_f)

    page = input1.getPage(0)
    print(page.cropBox)  # output:- RectangleObject([64, 492, 500, 692])
```

## configuration
 - Operating system:- `Ubuntu 20.04`
 - Python version:- `3.8.10`
 - PyMuPDF version:- `1.18.19`, installation method:- `from source`.

```
3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0] 
 linux 
 
PyMuPDF 1.18.19: Python bindings for the MuPDF 1.18.0 library.
Version date: 2021-09-16 16:45:29.
Built for Python 3.8 on linux (64-bit).
```

**Comment by JorjMcKie** [comment]
The coordinates are equal.
What do you get from `doc.xref_get_key(page.xref, "CorpBox")`?

**Comment by Kavan72** [comment]
If i using `doc.xref_get_key(page.xref, "CropBox")` i got the same CropBox which i have set but when i try to get CropBox from diff lib i'm getting diff then which i've used, and i've tried from two diff lib `PyPDF2`, `pdfplumber`

So basically if i set CropBox = (64, 100, 500, 300) after when i try to get CropBox using diff lib then i'm getting diff CropBox values

expected output:- 
 (64, 100, 500, 300)
 
 current output:- 
 [64, 492, 500, 692]

**Comment by JorjMcKie** [comment, marked as answer]
```python
>>> import fitz
>>> doc=fitz.open()
>>> page=doc.new_page()
>>> page.rect
Rect(0.0, 0.0, 595.0, 842.0)
>>> page.cropbox
Rect(0.0, 0.0, 595.0, 842.0)
>>> page.set_cropbox((100,100,400,400))
>>> page.cropbox
Rect(100.0, 100.0, 400.0, 400.0)
>>> doc.xref_get_key(page.xref, "CropBox")
('array', '[100 442 400 742]')
>>> print(doc.xref_object(page.xref))
<<
  /Type /Page
  /MediaBox [ 0 0 595 842 ]
  /Rotate 0
  /Resources 3 0 R
  /Parent 2 0 R
  /CropBox [ 100 442 400 742 ]  % <== this is what pypdf, etc return
>>
>>> page.rect
Rect(0.0, 0.0, 300.0, 300.0)
>>> 
```

**Comment by Kavan72** [comment]
ok understood, but one question is why it's giving diff then which i used on set_cropbox function ?
